### PR TITLE
[Test] Provid a frame agnostic API to test nn modules on both CPU and CUDA side.

### PR DESCRIPTION
**Pull request by classicsong**
## Description
1. make dgl.nn.xxx frame agnostic
2. make test.backend include dgl.nn modules
3. modify test_edge_softmax of test/mxnet/test_nn.py and
    test/pytorch/test_nn.py work on both CPU and GPU
4. Fix GPU initialization problem on nn.mxnet.Set2Set/GlobalAttentionPooling (by zihao)

#774 @yzh119  

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [ ] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR

### CODE REVIEW

**Comment by yzh119** [comment]
@classicsong is it ready for review?

**Comment by classicsong** [comment]
> @classicsong is it ready for review?

Yes

File: tests/backend/mxnet/__init__.py [hunk 1]
```diff
@@ -25,6 +25,9 @@ def allclose(a, b, rtol=1e-4, atol=1e-4):
 def randn(shape):
     return nd.random.randn(*shape)
 
+def rand(shape):
+    return nd.random.randint(0, 1, *shape)
```
Comment (yzh119) [inline tests/backend/mxnet/__init__.py]: Why randint?

Comment (classicsong) [reply tests/backend/mxnet/__init__.py]: > Why randint?

Oops. Just clean it out.
All test will use randn

**Pull request merged**